cpp/src/link_analysis/pagerank.cu

```diff
@@ -1,286 +1,278 @@
-
-  if (has_guess)
-  {
-    CUGRAPH_EXPECTS( pagerank != nullptr, "Column must be valid" );
-    copy<WT>(m, (WT*)pagerank, d_pr);
-  }
-
-  status = pagerankSolver<int32_t,WT>( m,nnz, (int*)graph->transposedAdjList->offsets->data, (int*)graph->transposedAdjList->indices->data, d_val,
-    personalization_subset, personalization_values, prsLen, has_personalization,
-    alpha, d_leaf_vector, has_guess, tolerance, max_iter, d_pr, residual);
-
-  if (status !=0)
-    switch ( status ) {
-      case -1: CUGRAPH_FAIL("Error : bad parameters in Pagerank");
-      case 1: CUGRAPH_FAIL("Warning : Pagerank did not reached the desired tolerance");
-      default:  CUGRAPH_FAIL("Pagerank exec failed");
-    }
-
-  copy<WT>(m, d_pr, (WT*)pagerank);
-
-  ALLOC_FREE_TRY(d_val, stream);
-#if 1/* temporary solution till https://github.com/NVlabs/cub/issues/162 is resolved */
-  CUDA_TRY(cudaFree(d_pr));
-#else
-  ALLOC_FREE_TRY(d_pr, stream);
-#endif
-  ALLOC_FREE_TRY(d_leaf_vector, stream);
-
-}
-}
-
-template <typename VT, typename WT>
-void pagerank(Graph *graph,WT* pagerank, size_t personalization_subset_size,
-  VT* personalization_subset,WT* personalization_values,
-  float alpha, float tolerance, int max_iter, bool has_guess) {
-  //
-  //  Pagerank operates on CSR and can't currently support 64-bit integers.
-  //  If csr doesn't exist, create it.  Then check type to make sure it is 32-bit.
-  //
-  CUGRAPH_EXPECTS(graph->transposedAdjList != nullptr, "Invalid API parameter: Transposed Adj List is needed by pagerank");
-
-  if (typeid(WT) != typeid(float) && typeid(WT) != typeid(double))
-    CUGRAPH_FAIL("Unsupported weight data type, please use float or double");
-  
-  if (typeid(VT) != typeid(int) )
-    CUGRAPH_FAIL("Unsupported personalization_subset data type, please use int");
-
-  CUGRAPH_EXPECTS( pagerank != nullptr , "Invalid API parameter: Pagerank array should be of size V" );
-
-  if (personalization_subset_size != 0) {
-    CUGRAPH_EXPECTS(typeid(VT) == typeid(int), "Unsupported personalization_subset data type");
-  }
-
-  return detail::pagerank_impl<VT,WT>(graph, pagerank,
-                                  personalization_subset_size, personalization_subset, personalization_values,
-                                  alpha, tolerance, max_iter, has_guess);
-}
-
-// explicit instantiation
-template void pagerank<int, float>(Graph *graph,float* pagerank,
-  size_t personalization_subset_size, int* personalization_subset,float* personalization_values,
-  float alpha, float tolerance, int max_iter, bool has_guess);
-template void pagerank<int, double>(Graph *graph,double* pagerank,
-  size_t personalization_subset_size, int* personalization_subset,double* personalization_values,
-  float alpha, float tolerance, int max_iter, bool has_guess);
-
-} //namespace cugraph 
\ No newline at end of file
+/*
```

Comment:
The git diff messed up. Have you changed encoding?
It is a bit hard to review the changes, would be great to fix.

Reply:
I did change the encoding.  This file was the only file I've encountered that is encoded as a DOS file.  I could switch it back if you like.

Reply:
Ugh.. ok. Let's keep it like this. It should not have been DOS in the first place.